test/test_torch.py

```diff
@@ -15389,6 +15389,88 @@ def test_minmax(self, device, dtype):
         self._test_minmax_helper(lambda x: torch._min_max(x)[0], np.min, device, dtype, skip_indices=True)
         self._test_minmax_helper(lambda x: torch._min_max(x)[1], np.max, device, dtype, skip_indices=True)
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @dtypes(*(torch.testing.get_all_int_dtypes() + [torch.bool]))
+    def test_maximum_minimum_int_and_bool(self, device, dtype):
+        ops = ((torch.maximum, np.maximum), (torch.minimum, np.minimum))
+        a_vals = (2, 3, 4, -2)
+        b_vals = (-1, 5, 2, 0)
+
+        for torch_op, numpy_op in ops:
+            a_tensor = torch.tensor(a_vals, device=device, dtype=dtype)
+            b_tensor = torch.tensor(b_vals, device=device, dtype=dtype)
+            tensor_result = torch_op(a_tensor, b_tensor)
+
+            out = torch.empty_like(a_tensor)
+            torch_op(a_tensor, b_tensor, out=out)
+
+            a_np = np.array(a_vals, dtype=torch_to_numpy_dtype_dict[dtype])
+            b_np = np.array(b_vals, dtype=torch_to_numpy_dtype_dict[dtype])
+            numpy_result = numpy_op(a_np, b_np)
+
+            self.assertEqual(tensor_result.cpu(), torch.from_numpy(numpy_result))
+            self.assertEqual(out.cpu(), torch.from_numpy(numpy_result))
+
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @dtypes(*(torch.testing.get_all_fp_dtypes()))
+    def test_maximum_minimum_float(self, device, dtype):
+        # np.maximum and np.minimum functions compare input arrays element-wisely.
+        # if one of the elements being compared is a NaN, then that element is returned.
+        ops = ((torch.maximum, np.maximum), (torch.minimum, np.minimum))
+        a_vals = (float('inf'), -float('inf'), 2.0, float('nan'), -1.0, float('nan'))
```

Comment:
You could also simplify this test using `np.randn` to generate a float tensor, if you like. You would still have to generate an "extremal values" test case with -inf, inf, and nan by hand, though. You also can simplify by just comparing the torch and numpy results. You don't need to assert the results are correct. 

To handle bfloat16 you can just cast the generated arrays to the desired torch type (leaving the arrays in double) and then compare using `self.assertEqual(actual, expected, exact_dtype=False)`.

Reply:
Thanks for the suggestions, I have split the test cases into two methods
- test_maximum_minimum_float: use `np.randn` to generate random float inputs.
- test_maximum_minimum_float_nan_and_inf: test external values like `nan`, `-inf` and `inf`

For `bfloat16`,  I have to set rtol and atol like below to prevent precision issues 
```
self.assertEqual(tensor_result, numpy_result, rtol=0.016, atol=1e-5, exact_dtype=False)
````
The values of rtol and atol refers to `common_utils.py` file.
https://github.com/pytorch/pytorch/blob/42b4a7132e7c6f1df963b473d1583e4791fb1808/torch/testing/_internal/common_utils.py#L939-L942